Firebase/Database/Api/FIRDatabase.m

```diff
@@ -38,142 +41,27 @@ @interface FIRDatabase ()
-                  usingBlock:^(NSNotification * _Nonnull note) {
-      NSString *appName = note.userInfo[kFIRAppNameKey];
-      if (appName == nil) { return; }
-
-      FIRDatabaseDictionary* instances = [self instances];
-      @synchronized (instances) {
-          NSMutableDictionary<FRepoInfo *, FIRDatabase *> *databaseInstances = instances[appName];
-          if (databaseInstances) {
-              // Clean up the deleted instance in an effort to remove any resources still in use.
-              // Note: Any leftover instances of this exact database will be invalid.
-              for (FIRDatabase * database in [databaseInstances allValues]) {
-                  [FRepoManager disposeRepos:database.config];
-              }
-              [instances removeObjectForKey:appName];
-          }
-      }
-  }];
-}
-
-/**
- * A static NSMutableDictionary of FirebaseApp name and FRepoInfo to
- * FirebaseDatabase instance. To ensure thread-safety, it should only be
- * accessed in databaseForApp:URL:, which is synchronized.
- *
- * TODO: This serves a duplicate purpose as RepoManager.  We should clean up.
- * TODO: We should maybe be conscious of leaks and make this a weak map or
- * similar but we have a lot of work to do to allow FirebaseDatabase/Repo etc.
- * to be GC'd.
- */
-+ (FIRDatabaseDictionary *)instances {
-    static dispatch_once_t pred = 0;
-    static FIRDatabaseDictionary *instances;
-    dispatch_once(&pred, ^{
-        instances = [NSMutableDictionary dictionary];
-    });
-    return instances;
-}
-
 + (FIRDatabase *)database {
-    if (![FIRApp isDefaultAppConfigured]) {
-        [NSException raise:@"FIRAppNotConfigured"
-                    format:@"Failed to get default Firebase Database instance. Must call `[FIRApp "
-                           @"configure]` (`FirebaseApp.configure()` in Swift) before using "
-                           @"Firebase Database."];
-    }
-    FIRApp *app = [FIRApp defaultApp];
-    return [FIRDatabase databaseForApp:app];
+    return [FIRDatabase databaseForApp:[FIRApp defaultApp]];
 }
 
 + (FIRDatabase *)databaseWithURL:(NSString *)url {
-    FIRApp *app = [FIRApp defaultApp];
-    if (app == nil) {
-        [NSException raise:@"FIRAppNotConfigured"
-                    format:@"Failed to get default Firebase Database instance. "
-                           @"Must call `[FIRApp configure]` (`FirebaseApp.configure()` in Swift) "
-                           @"before using Firebase Database."];
-    }
-    return [FIRDatabase databaseForApp:app URL:url];
+    return [FIRDatabase databaseForApp:[FIRApp defaultApp] URL:url];
 }
 
 + (FIRDatabase *)databaseForApp:(FIRApp *)app {
-    if (app == nil) {
-        [NSException raise:@"InvalidFIRApp" format:@"nil FIRApp instance passed to databaseForApp."];
-    }
-
     return [FIRDatabase databaseForApp:app URL:app.options.databaseURL];
 }
 
 + (FIRDatabase *)databaseForApp:(FIRApp *)app URL:(NSString *)url {
-    if (app == nil) {
```

Comment:
I think it's still okay to validate the app here, since otherwise the `app.container` will be nil and won't actually log any errors.

Reply:
Done